crates/mem-pool/src/pool.rs

```diff
@@ -914,11 +914,35 @@ impl MemPool {
         assert!(self.mem_block.finalized_custodians().is_none());
         assert!(self.mem_block.txs().is_empty());
 
+        fn filter_withdrawals(
+            state: &MemStateTree<'_>,
+            withdrawal: &WithdrawalRequestExtra,
+        ) -> bool {
+            let id = state
+                .get_account_id_by_script_hash(&withdrawal.raw().account_script_hash().unpack())
+                .expect("get id")
+                .expect("id exist");
+            let nonce = state.get_nonce(id).expect("get nonce");
+            let expected_nonce: u32 = withdrawal.raw().nonce().unpack();
+            // ignore withdrawal mismatch the nonce
+            nonce == expected_nonce
```

Comment:
What will happen IF `withdrawal.nonce < state.get_nonce(id)`? Is it possible?

Reply:
Does the mem-pool check the nonce of a withdrawal when it receives one? If so, how does a withdrawal with a mismatched nonce enter the mem-pool?

Reply:
Now I get it. Thanks for explaining.